Update for pre-plan run tasks beta (#72)
hashicorp/terraform-docs-common: website/docs/cloud-docs/run/states.mdx

_States in this stage:_

- **Fetching:** If Terraform Cloud has not yet fetched the configuration from VCS, the run will go into this state until the configuration is available.

_Leaving this stage:_

- If Terraform Cloud encounters an error when fetching the configuration from VCS, the run does not continue (**Plan Errored** state).
- If the fetching was successful, the run moves to the next stage.

## 3. The Pre-Plan Stage

The pre-plan phase executes any configured run tasks before the plan is created, so it only occurs if there are [run tasks](/cloud-docs/workspaces/settings/run-tasks) enabled
for the workspace. All runs can enter this phase, including [speculative plans](/cloud-docs/run/#speculative-plans). During this phase, Terraform Cloud sends information about the run to the configured external system and waits for a `passed` or `failed` response to determine whether the run can continue.

-> **Note:** The information sent to the configured external system includes the configuration version of the run.

_States in this stage:_

- **Pre-plan running:** Terraform Cloud is waiting for a response from the configured external system(s).

>   - External systems must respond initially with a `200 OK` acknowledging the request is in progress. After that, they have 10 minutes to return a status of `passed`, `running`, or `failed`, or the timeout will expire and the task will be assumed to be in the `failed` status.

Reviewer: ```suggestion
 
 External systems must respond initially with a `200 OK` acknowledging the request is in progress. After that, they have 10 minutes to return a status of `passed`, `running`, or `failed`. If the timeout expires, Terraform Cloud assumes that the run tasks is in the `failed` status.
```

_States in this stage:_

- **Fetching:** If Terraform Cloud has not yet fetched the configuration from VCS, the run will go into this state until the configuration is available.

_Leaving this stage:_

- If Terraform Cloud encounters an error when fetching the configuration from VCS, the run does not continue (**Plan Errored** state).
- If Terraform successfully fetches the configuration, the run moves to the next stage.

## 3. The Pre-Plan Stage

The pre-plan phase only occurs if there are enabled [run tasks](/cloud-docs/workspaces/settings/run-tasks) in the workspace that are configured to begin before Terraform creates the plan. Terraform Cloud sends information about the run to the configured external system and waits for a `passed` or `failed` response to determine whether the run can continue. The information sent to the external system includes the configuration version of the run.

All runs can enter this phase, including [speculative plans](/cloud-docs/run/#speculative-plans). 

_States in this stage:_

- **Pre-plan running:** Terraform Cloud is waiting for a response from the configured external system(s).
 

>  External systems must respond initially with a `200 OK` acknowledging the request is in progress. After that, they have 10 minutes to return a status of `passed`, `running`, or `failed`. If the timeout expires, Terraform Cloud assumes that the run tasks is in the `failed` status.

Reviewer 2: Are we going to repeat ourselves a few times on this page?

Reviewer 2: Also, the other time this is mentioned seems to be a unordered list:
https://github.com/hashicorp/terraform-docs-common/pull/72/files#diff-9c4b29067fbb525b780309e425de54e31d50d132eb3dcd46ca739a137277b057R90

Author: I've fixed this up to be consistent (unorderd list)

However I have no answer to the repeating bit.  It may be a necessary evil.
